src/client/oldCLient.cpp

```diff
@@ -0,0 +1,240 @@
+#include <cstddef>
+#include <WinSock2.h>
+#include <winsock.h>
+#include <WS2tcpip.h>
+#include <stdio.h>
+#include <Windows.h>
+#include <iostream>
+#include <string>
+using namespace std;
+#define DEFAULT_PORT "6667"
+#define DEFAULT_BUFLEN 2048
+#define URL "testnet.ergo.chat"
+
+
+struct addrinfo *ptr = NULL;
+struct addrinfo hints;
+
+int main()
+{
+
+    WSADATA wsaData;
+
+    int iResult;
+
+    // Initialize Winsock
+    iResult = WSAStartup(MAKEWORD(2, 1), &wsaData);
+
+    ZeroMemory(&hints, sizeof(hints));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_protocol = IPPROTO_TCP;
+
+    iResult = getaddrinfo(URL, DEFAULT_PORT, &hints, &ptr);
+
+    SOCKET ConnectSocket = INVALID_SOCKET;
+
+    ConnectSocket = socket(ptr->ai_family, ptr->ai_socktype,
+                           ptr->ai_protocol);
+
+    // Connect to server.
+    iResult = connect(ConnectSocket, ptr->ai_addr, (int)ptr->ai_addrlen);
+
+    freeaddrinfo(ptr);
```

Comment:
C'est bien en général, je ferais juste un null check sur freeaddrinfo(ptr) juste en cas de sécurité, car on définit ptr en tant que NULL.

```suggestion
if (ptr != NULL)
{
    freeaddrinfo(ptr);
}
```